src/tokens.md

```diff
@@ -328,6 +330,76 @@ b"\x52"; b"R"; br"R";                // R
 b"\\x52"; br"\x52";                  // \x52
 ```
 
+### C string and raw C string literals
+
+#### C string literals
+
+> **<sup>Lexer</sup>**\
+> C_STRING_LITERAL :\
+> &nbsp;&nbsp; `c"` ( ASCII_FOR_C_STRING | BYTE_ESCAPE | STRING_CONTINUE )<sup>\*</sup> `"` SUFFIX<sup>?</sup>
+>
+> ASCII_FOR_C_STRING :\
+> &nbsp;&nbsp; _any non-NUL ASCII (i.e 0x01 to 0x7F), except_ `"`, `\` _and IsolatedCR_
+
+A non-raw _C string literal_ is a sequence of ASCII characters and _escapes_,
+preceded by the characters `U+0063` (`c`) and `U+0022` (double-quote), and
+followed by the character `U+0022`. If the character `U+0022` is present within
+the literal, it must be _escaped_ by a preceding `U+005C` (`\`) character.
+Alternatively, a C string literal can be a _raw C string literal_, defined
+below. The type of a C string literal is `&core::ffi::CStr`.
+
+Some additional _escapes_ are available in either C or non-raw C string
+literals. An escape starts with a `U+005C` (`\`) and continues with one of the
+following forms:
+
+* A _byte escape_ escape starts with `U+0078` (`x`) and is followed by exactly
+  two _hex digits_. It denotes the byte equal to the provided hex value. The
+  byte escape sequence `\x00` is forbidden, as C strings may not contain `NUL`.
```

Comment:
I might reword this to cover both a bare NUL and `\x00` are not allowed.

Reply:
Done-ish -- I moved the section on `NUL` escapes to the bottom of this section, to make it clearer that `\0`, `\x00`, and `\u{0000}` are treated uniformly.